mnelab/mainwindow.py

```diff
@@ -494,12 +496,23 @@ def run_ica(self):
             method = dialog.method.currentText()
             exclude_bad_segments = dialog.exclude_bad_segments.isChecked()
             fit_params = {}
+            self.model.history.append("fit_params = {}")
```

Comment:
Instead of extending an empty dict, you could add the finished `fit_params` to the history after everything has been constructed.